src/main/scala/treadle/VcdReplayTester.scala

```diff
@@ -82,61 +91,90 @@ class VcdReplayTester(optionsManager: VcdReplayTesterOptions) extends LazyLoggin
+trait VcdReplayTesterCli { this: Shell =>
+  parser.note("VcdReplayTester Front End Options")
 
-    if (optionsManager.parse(args)) {
-      val repl = new VcdReplayTester(optionsManager)
-      repl.run()
-    }
+
+}
+
+class VcdReplayTesterStage extends Stage {
+  val shell = new Shell("vcd_replay_tester") with VcdReplayTesterCli with FirrtlCli
+
+  def apply(annotationSeq: AnnotationSeq): VcdReplayTester = {
+    new VcdReplayTester(annotationSeq)
+  }
+
+  override def run(annotations: AnnotationSeq): AnnotationSeq = {
+    new VcdReplayTester(annotations)
+    annotations
   }
 }
 
-case class VcdReplayOptions(
-    firrtlSourceName:     String = "",
-    vcdSourceName:        String = "",
-    skipEvents:           Int = 0,
-    eventsToRun:          Int = -1,
-    testAliasedWires:     Boolean = false)
-  extends firrtl.ComposableOptions
-
-trait HasVcdReplayOptions {
-  self: ExecutionOptionsManager =>
-
-  var goldenVcdOptions = VcdReplayOptions()
-
-  parser.note("golden-vcd")
-
-  parser.opt[String]("firrtl-source")
-    .abbr("fs")
-    .valueName("<firrtl-source-file>")
-    .foreach { x => goldenVcdOptions = goldenVcdOptions.copy(firrtlSourceName = x) }
-    .text("firrtl source file to load on startup")
-
-  parser.opt[String]("vcd-file")
-    .abbr("vcd")
-    .valueName("<vcd-file>")
-    .foreach { x => goldenVcdOptions = goldenVcdOptions.copy(vcdSourceName = x) }
-    .text("firrtl source file to load on startup")
-
-  parser.opt[Int]("skip-events")
-    .abbr("se")
-    .valueName("<number>")
-    .foreach { x => goldenVcdOptions = goldenVcdOptions.copy(skipEvents = x) }
-    .text("number of events to skip before starting")
-
-  parser.opt[Int]("events-to-run")
-    .abbr("etr")
-    .valueName("<number>")
-    .foreach { x => goldenVcdOptions = goldenVcdOptions.copy(eventsToRun = x) }
-    .text("number of events to run")
-
-  parser.opt[Unit]("test-aliased-wires")
-    .abbr("taw")
-    .foreach { _ => goldenVcdOptions = goldenVcdOptions.copy(testAliasedWires = true) }
-    .text("number of events to run")
+object VcdReplayTester extends StageMain(new VcdReplayTesterStage)
+
+sealed trait VcdReplayTesterOptions extends Unserializable { this: Annotation => }
+
+case class VcdReplayVcdFile(fileName: String) extends NoTargetAnnotation with VcdReplayTesterOptions
+
+case class VcdReplaySkipEvents(count: Int) extends NoTargetAnnotation with VcdReplayTesterOptions
+
+case class VcdReplayEventsToRun(count: Int) extends NoTargetAnnotation with VcdReplayTesterOptions
+
+/**
+  * makes a more convenient way of specifying firrtl source
+  */
+case object VcdReplayFirrtlSource extends NoTargetAnnotation with VcdReplayTesterOptions with HasShellOptions {
+  val options = Seq(
+    new ShellOption[String](
+      longOption = "firrtl-source",
+      shortOption = Some("fs"),
+      toAnnotationSeq = (s: String) => Seq(FirrtlSourceAnnotation(s)),
+      helpText = "firrtl source file to load on startup"
+    )
```

Comment:
If you want, you can use `helpValueName: Option[String]` to provide the same `<firrtl-source-file>`. Do you think it makes sense to include those? Ditto for the ones below.